src/test/clip.t.sol

```diff
@@ -49,6 +49,28 @@ contract Guy {
     function bark(Dog dog, bytes32 ilk, address urn) external {
         dog.bark(ilk, urn);
     }
+
+    function clipperCall(uint256 owe, uint256 slice, bytes calldata data)
+        external virtual {
+        Vat(address(clip.vat())).suck(address(0), address(this), owe);
```

Comment:
The test works, but maybe as an actual example I'd go for something closer to the reality where you can actually test that you are receiving the collateral up front.
So we'd need a contract that exchanges your collateral for DAI and you can leave a small profit to show the case of why someone go run it. It'd be a plain arbitrage operation.

Reply:
Updated with a more realistic example